### PR TITLE
Feature 3 models

### PR DESCRIPTION
**Pull request by mattChrisP**
After doing the point transfer, Transaction model is updated with the info of the transfers, but I didn't manage to update the loyalty program data itself. I need help with this cause I have debugged for quite a long time but didn't manage to solve it. You can see the exchange points page by going to /loyalty_program_data path. For clarity, every user will have all loyalty programs with the initial amount of points equal to zero. And the points is transferred from the bank account money of the user, not with respect to other loyalty program. So there is no point exchange between loyalty programs @ExtremelySunnyYK @suvasan19

### CODE REVIEW

File: app/controllers/loyalty_program_data_controller.rb [hunk 1]
```diff
@@ -3,11 +3,13 @@ class LoyaltyProgramDataController < ApplicationController
 
   # GET /loyalty_program_data or /loyalty_program_data.json
   def index
-    @loyalty_program_data = LoyaltyProgramDatum.all
-    @transaction = Transaction.all
+    @loyalty_program_data = LoyaltyProgramDatum.where(account_id: current_user)
```
Comment (ExtremelySunnyYK) [inline app/controllers/loyalty_program_data_controller.rb:6]: Did you test this out? seems like there might be error here. 

 should it be current_user.account.id ?

File: app/controllers/loyalty_program_data_controller.rb [hunk 2]
```diff
@@ -21,6 +23,21 @@ def new
...
+        
+    #   else
+    #     form_complete = false
+    #   end
+      
+      
+    #   if form_complete
+    #     Transaction.create(amount: params[:points], loyalty_program_data_id: 100, status: 0)
+    #   end
+    # end
```
Comment (ExtremelySunnyYK) [inline app/controllers/loyalty_program_data_controller.rb:40]: why are the lines commented out here?

File: app/controllers/transactions_controller.rb [hunk 3]
```diff
@@ -3,7 +3,7 @@ class TransactionsController < ApplicationController
 
   # GET /transactions or /transactions.json
   def index
-    @transactions = Transaction.all
+    @transactions = Transaction.where(account_id: current_user)
```
Comment (ExtremelySunnyYK) [inline app/controllers/transactions_controller.rb:6]: do we need to do current_user.id?

File: app/controllers/loyalty_program_data_controller.rb [hunk 4]
```diff
@@ -42,6 +59,13 @@ def create
   def update
     respond_to do |format|
       if @loyalty_program_datum.update(loyalty_program_datum_params)
+
+
+        # I want to add points to a particular loyalty program data with loyalty program id = params[:id], but dont know how to do
+        # @loyalty_program_datum.where(loyalty_program_id: params[:id]).points += loyalty_program_datum_params[:points].to_d
+
+ 
+        Transaction.create(amount: loyalty_program_datum_params[:points].to_d, loyalty_program_data_id: params[:id], status: 0, account_id: current_user.id)
```
Comment (ExtremelySunnyYK) [inline app/controllers/loyalty_program_data_controller.rb:68]: should be account id of current_user?

**Comment by ExtremelySunnyYK** [comment]
@mattChrisP maybe we can merge this in first?

**Comment by mattChrisP** [comment]
ok hold on I will push some stuff first

**Pull request merged**